.github/workflows/test.yml

```diff
@@ -38,3 +38,50 @@ jobs:
           nix flake lock
           nix run .#updateMaterialized
           git diff --exit-code
+  shared-node-stub:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      CLIENT_CERT: ${{ secrets.DEV_CI_CLIENT_CERT }}
+      SERVER_CERT: ${{ secrets.DEV_CI_SERVER_CERT }}
+      TREASURY_ADDR: ${{ secrets.TREASURY_ADDR }}
+      TREASURY_VKEY: ${{ secrets.TREASURY_VKEY }}
+      TREASURY_SKEY: ${{ secrets.TREASURY_SKEY }}
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - uses: cachix/install-nix-action@v17
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          extra_nix_config: |
+            allowed-uris = ${{ env.ALLOWED_URIS }}
+            trusted-public-keys = ${{ env.TRUSTED_PUBLIC_KEYS }}
+            substituters = ${{ env.SUBSTITUTERS }}
+            experimental-features = nix-command flakes
+      - uses: cachix/cachix-action@v10
+        with:
+          name: ${{ env.CACHE_NAME }}
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      - run: |
+          cat > client.pem <<EOF
+          $CLIENT_CERT
+          EOF
+          cat > server.crt <<EOF
+          $SERVER_CERT
+          EOF
+          mkdir keys
+          cat > keys/payment.addr <<EOF
+          $TREASURY_ADDR
+          EOF
+          cat > keys/payment.vkey <<EOF
+          $TREASURY_VKEY
+          EOF
+          cat > keys/payment.skey <<EOF
+          $TREASURY_SKEY
+          EOF
+          nix develop -p socat --command "socat UNIX-LISTEN:node.sock,fork OPENSSL-CONNECT:dev-ci-node.marlowe-finance.io:443,cert=client.pem,cafile=server.crt,openssl-commonname=marlowe-node-server,snihost=dev-ci-node.marlowe-finance.io" &>socat-log &
+          export CARDANO_NODE_SOCKET_PATH=node.sock
+          export CARDANO_TESTNET_MAGIC=1566
+          export TREASURY=keys/
+          nix develop --command "cabal install exe:marlowe-cli && cd marlowe-cli && . ./run-nonpab-tests.sh"
```

Comment:
Might not need to install cabal part of this line as nix develop already install the marlowe cli for us.